### PR TITLE
TEC-4882 - Add pro sponsor level and speaker group reorder to settings.

### PR DESCRIPTION
**Pull request by jesseeproductions**
- Adds Sponsor Level and Speaker Group to Settings

Artifact:
-Settings with Reorder - https://cloudup.com/c66LGsdl598

_Ref:_ [TEC-4882](https://theeventscalendar.atlassian.net/browse/TEC-4882)

[TEC-4882]: https://theeventscalendar.atlassian.net/browse/TEC-4882?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Conference/Admin/Settings.php [hunk 1]
```diff
@@ -69,6 +71,14 @@ public function init() {
 
 		// Register sponsor page URL field in the "wpcs_section_info" section, inside the "wpcs" page.
 		add_settings_field( 'wpcsp_field_sponsor_page_url', _x( 'Sponsor URL Redirect', 'settings field title', 'wp-conference-schedule' ), [ $this, 'field_sponsor_page_url_cb' ], 'wpcs', 'wpcs_section_settings' );
+
+		register_setting( 'wpcs', 'wpcsp_conference_sponsor_level_order', array( $this, 'validate_sponsor_options' ) );
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
		register_setting( 'wpcs', 'wpcsp_conference_sponsor_level_order', [ $this, 'validate_sponsor_options' ] );
```

File: src/Conference/Admin/Settings.php [hunk 2]
```diff
@@ -69,6 +71,14 @@ public function init() {
...
+
+		add_settings_field( 'wpcsp_field_sponsor_level_order', _x( 'Sponsor Level Order', 'settings field title', 'wp-conference-schedule' ), [ $this, 'render_order_sponsor_levels' ], 'wpcs', 'wpcs_section_settings' );
+
+		register_setting( 'wpcs', 'wpcsp_speaker_level_order', array( $this, 'validate_sponsor_options' ) );
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
		register_setting( 'wpcs', 'wpcsp_speaker_level_order', [ $this, 'validate_sponsor_options' ] );
```

File: src/Conference/Admin/Settings.php [hunk 3]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
 		<?php
 	}
+
+	/**
+	 * Renders the Order Sponsor Levels admin page.
+	 *
+	 * @since TBD
+	 */
+	public function render_order_sponsor_levels() {
+	    $this->render_order_levels('wpcsp_conference_sponsor_level_order', Plugin::SPONSOR_LEVEL_TAXONOMY, _x('Change the order of sponsor levels displayed in the sponsors page template.', 'Directions for ordering of sponsor levels in the settings.', 'wp-conference-schedule'));
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
		$this->render_order_levels(
			'wpcsp_conference_sponsor_level_order',
			Plugin::SPONSOR_LEVEL_TAXONOMY,
			_x( 'Change the order of sponsor levels displayed in the sponsors page template.', 'Directions for ordering of sponsor levels in the settings.', 'wp-conference-schedule' )
		);
```

File: src/Conference/Admin/Settings.php [hunk 4]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
+	}
+
+	/**
+	 * Renders the Order Speaker Levels admin page.
+	 *
+	 * @since TBD
+	 */
+	public function render_order_speaker_levels() {
+	    $this->render_order_levels('wpcsp_speaker_level_order', Plugin::GROUP_TAXONOMY, _x('Change the order of speaker groups displayed in the speaker page template.', 'Directions for ordering of speaker groups in the settings.', 'wp-conference-schedule'));
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
		$this->render_order_levels(
			'wpcsp_speaker_level_order', 
			Plugin::GROUP_TAXONOMY, 
			_x('Change the order of speaker groups displayed in the speaker page template.', 'Directions for ordering of speaker groups in the settings.', 'wp-conference-schedule' ) 
		);
```

File: src/Conference/Admin/Settings.php [hunk 5]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
+	 *
+	 * @return array Array of term objects.
+	 */
+	public function get_sponsor_levels(string $option, string $taxonomy): array {
+	    $option = get_option($option, []);
+	    $term_objects = get_terms($taxonomy, ['get' => 'all']);
+	    $terms = [];
+
+	    foreach ($term_objects as $term) {
+	        $terms[$term->term_id] = $term;
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
	        $terms[ $term->term_id ] = $term;
```

File: src/Conference/Admin/Settings.php [hunk 6]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
+	 * @param string $taxonomy The taxonomy to fetch terms for.
+	 *
+	 * @return array Array of term objects.
+	 */
+	public function get_sponsor_levels(string $option, string $taxonomy): array {
+	    $option = get_option($option, []);
+	    $term_objects = get_terms($taxonomy, ['get' => 'all']);
+	    $terms = [];
+
+	    foreach ($term_objects as $term) {
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
	    foreach ( $term_objects as $term ) {
```

File: src/Conference/Admin/Settings.php [hunk 7]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
+	 * @since TBD
+	 *
+	 * @param string $option The option key to fetch from the database.
+	 * @param string $taxonomy The taxonomy to fetch terms for.
+	 *
+	 * @return array Array of term objects.
+	 */
+	public function get_sponsor_levels(string $option, string $taxonomy): array {
+	    $option = get_option($option, []);
+	    $term_objects = get_terms($taxonomy, ['get' => 'all']);
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
	    $term_objects = get_terms( $taxonomy, [ 'get' => 'all' ] );
```

File: src/Conference/Admin/Settings.php [hunk 8]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
+	 *
+	 * @since TBD
+	 *
+	 * @param string $option The option key to fetch from the database.
+	 * @param string $taxonomy The taxonomy to fetch terms for.
+	 *
+	 * @return array Array of term objects.
+	 */
+	public function get_sponsor_levels(string $option, string $taxonomy): array {
+	    $option = get_option($option, []);
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
	    $option = get_option( $option, [] );
```

File: src/Conference/Admin/Settings.php [hunk 9]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
+	 * Returns the sponsor level terms in set order.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $option The option key to fetch from the database.
+	 * @param string $taxonomy The taxonomy to fetch terms for.
+	 *
+	 * @return array Array of term objects.
+	 */
+	public function get_sponsor_levels(string $option, string $taxonomy): array {
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
	public function get_sponsor_levels( string $option, string $taxonomy ): array {
```

File: src/Conference/Admin/Settings.php [hunk 10]
```diff
@@ -233,4 +243,117 @@ public function options_page_html() {
...
+	 * @param array|null $input The input options array.
+	 *
+	 * @return array|null Sanitized array of integers or null.
+	 */
+	public function validate_sponsor_options(?array $input): ?array {
+	    if (!is_array($input)) {
+	        return null;
+	    }
+
+	    return array_values(array_map('intval', $input));
```
Comment (bordoni) [inline src/Conference/Admin/Settings.php]:
```suggestion
	public function validate_sponsor_options( ?array $input ): ?array {
	    if ( ! is_array( $input ) ) {
	        return null;
	    }

	    return array_values( array_map( 'intval', $input ) );
```

**Review by bordoni — CHANGES_REQUESTED** [review]
Most of the file has spaces mixed with tabs for indentation. Lets fix that too.

File: src/Conference/Admin/Settings.php (hunk 10 above)
Comment (jesseeproductions) [reply src/Conference/Admin/Settings.php]: I fixed the spacing. Let me know if there is something else. Thanks.

**Review by bordoni — APPROVED** [review]

**Pull request merged**